### PR TITLE
Added testing

### PR DESCRIPTION
**Pull request by jarovo**
I've addressed the review comments.

### CODE REVIEW

File: tests/test_container_provider.py [hunk 1]
```diff
@@ -7,6 +8,16 @@
 import manageiq
 
 
+provider_name = "Provider name 1 with some unicode characters «ταБЬℓσ»"
+provider_hostname = "some-provider-hostname.tld"
```
Comment (dkorn) [inline tests/test_container_provider.py]: shouldn't these constants be all capital letters?

File: tests/test_container_provider.py [hunk 2]
```diff
@@ -7,6 +8,16 @@
...
+provider_token = "THE_PROVIDER_TOKEN"
+provider_port = "8443"
+provider_id = 134
+hawkular_hostname = "some-hawkular-hostname.tld"
+hawkular_port = "54321"
```
Comment (dkorn) [inline tests/test_container_provider.py]: perhaps this can also be a "real" port example, like "443"?

File: tests/test_container_provider.py [hunk 3]
```diff
@@ -40,82 +52,100 @@ class AnsibleModuleFailed(Exception):
 
 @pytest.fixture()
 def miq(miq_api_class, miq_ansible_module, the_provider):
+
+    # TODO(jaryn) This would ideally be inside test cases that need it.
```
Comment (dkorn) [inline tests/test_container_provider.py]: IINM, this is possible after https://github.com/dkorn/manageiq-ansible-module/pull/15 is merged right?

File: tests/test_container_provider.py [hunk 4]
```diff
@@ -40,82 +52,100 @@ class AnsibleModuleFailed(Exception):
...
     def fail(msg):
         raise AnsibleModuleFailed(msg)
 
     miq_ansible_module.fail_json = fail
-    miq = manageiq.ManageIQ(miq_ansible_module, "http://themanageiq.tld",
+    miq = manageiq.ManageIQ(miq_ansible_module, manageiq_hostname,
                             "The username", "The password")
 
-    miq_api_class.return_value.post.return_value = dict(results=[dict(id=0)])
+    miq_api_class.return_value.post.return_value = dict(results=[
```
Comment (dkorn) [inline tests/test_container_provider.py:64]: can we create another post return value for a delete action? (returns different fields)

**Review by dkorn — CHANGES_REQUESTED** [review]
needs a rebase and few inline comments/questions.

File: tests/test_container_provider.py (hunk 4 above)
Comment (jarovo) [reply tests/test_container_provider.py:64]: The MIQ is using HTTP POST to do delete? Hmm Why doesn't it use DELETE instead?

File: tests/test_container_provider.py (hunk 3 above)
Comment (jarovo) [reply tests/test_container_provider.py]: removed

File: tests/test_container_provider.py (hunk 2 above)
Comment (jarovo) [reply tests/test_container_provider.py]: done

File: tests/test_container_provider.py (hunk 1 above)
Comment (jarovo) [reply tests/test_container_provider.py]: they are now

File: tests/test_container_provider.py (hunk 4 above)
Comment (dkorn) [reply tests/test_container_provider.py:64]: they also support DELETE but IINM when using DELETE you don't get a response (which we need to know that it was successful and track the deletion task).
Do you have an idea how we can support more than one post?

File: tests/test_container_provider.py (hunk 1 above)
Comment (dkorn) [reply tests/test_container_provider.py]: :+1:

File: tests/test_container_provider.py (hunk 2 above)
Comment (dkorn) [reply tests/test_container_provider.py]: :+1:

File: tests/test_container_provider.py (hunk 4 above)
Comment (jarovo) [reply tests/test_container_provider.py:64]: Interesting.

I think we gonna need another mock for the different post.

 Anyway please merge as I think it is not a topic of this PR

**Review by dkorn — APPROVED** [review]

**Pull request merged**